cobra/core/model.py

```diff
@@ -414,37 +415,119 @@ def add_reactions(self, reaction_list):
+        problem and solving it. Cobrapy uses the optlang package to
+        accomplish that and with this property you can get access to the
+        problem interface directly.
+
+        Returns
+        -------
+        optlang.interface
+            The problem interface that defines methods for interacting with
+            the problem and associated solver directly.
+        """
+        return self.solver.interface
+
+    @property
+    def variables(self):
+        """The mathematical variables in the cobra model.
+
+        In a cobra model, most variables are reactions. However,
+        for specific use cases, it may also be useful to have other types of
+        variables. This property defines all variables currently associated
+        with the model's problem.
+
+        Returns
+        -------
+        optlang.container.Container
+            A container with all associated variables.
+        """
+        return self.solver.variables
+
+    @property
+    def constraints(self):
+        """The constraints in the cobra model.
+
+        In a cobra model, most constraints are metabolites and their
+        stoichiometries. However, for specific use cases, it may also be
+        useful to have other types of constraints. This property defines all
+        constraints currently associated with the model's problem.
+
+        Returns
+        -------
+        optlang.container.Container
+            A container with all associated constraints.
+        """
+        return self.solver.constraints
+
     def _populate_solver(self, reaction_list, metabolite_list=None):
         """Populate attached solver with constraints and variables that
         model the provided reactions.
         """
         constraint_terms = AutoVivification()
         if metabolite_list is not None:
             for met in metabolite_list:
-                constraint = self.solver.interface.Constraint(S.Zero,
-                                                              name=met.id,
-                                                              lb=0, ub=0)
+                constraint = self.problem.Constraint(S.Zero,
+                                                     name=met.id,
+                                                     lb=0, ub=0)
                 self.solver.add(constraint)
 
         for reaction in reaction_list:
 
             reverse_lb, reverse_ub, forward_lb, forward_ub = \
                 separate_forward_and_reverse_bounds(*reaction.bounds)
 
-            forward_variable = self.solver.interface.Variable(
+            forward_variable = self.problem.Variable(
                 reaction.id, lb=forward_lb, ub=forward_ub)
-            reverse_variable = self.solver.interface.Variable(
+            reverse_variable = self.problem.Variable(
                 reaction._get_reverse_id(), lb=reverse_lb, ub=reverse_ub)
 
             self.solver.add(forward_variable)
```

Comment:
Yes, updated and merged with your pr, now we use model.add_cons_vars where possible